### PR TITLE
BOM-2640: Upgrade & Codemods

### PR DESCRIPTION
**Pull request by mzulqarnain1**
Relevant JIRA : https://openedx.atlassian.net/browse/BOM-2640

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+name: Python CI
+
+on:
+  push:
+    branches:
+    - master
+    - /^\d+\.\d+(\.\d+)?(-\S*)?$/
```
Comment (mraarif) [inline .github/workflows/ci.yml:7]: shouldn't this run on `master`

**Review by mraarif — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mzulqarnain1) [reply .github/workflows/ci.yml:7]: I used the same regex from the Travis file.

**Review by Jawayria — APPROVED** [review]

**Review by UsamaSadiq — APPROVED** [review]

**Comment by UsamaSadiq** [comment]
Looks good to go. Most of the changes are quality fixes which are needed for the quality build to pass with new pylint version so these changes need to be in same PR.

**Comment by UsamaSadiq** [comment]
@jmbowman 
Merging this PR will add `Django 3.2` support in `ecommerce-worker` but we won't be able to remove the `Django<2.3` constraint and upgrade to `Django 3.2` unless `ecommerce` has been upgraded to `Django 3.2` since this is also being used in `ecommerce` as a package.

**Pull request merged**